### ISSUE TITLE
Migrating on existing tenants in Production

### ISSUE THREAD

**Issue by churcho** [ateliware/triplex#66]
How do you run migrations in production environments with no `mix`?

I made changes to some of my tables and would like to propagate this to already existing tenants.

Do we need to keep a table with existing [tenants]?(https://github.com/ateliware/triplex/blob/f49c43fc41169818a1454b3eb3ff5c3bc4815e4d/lib/triplex/config.ex#L19)

**Comment by kelvinst** [comment]
@churcho sorry it took so long, what do you mean? `mix triplex.migrate` should do that for your (running the migrations on all existing tenants). Please checkout the docs for it: https://hexdocs.pm/triplex/Mix.Tasks.Triplex.Migrate.html#content

**Comment by kelvinst** [comment]
If you use release though, that a little bit trickier, but you can combine [`Triplex.all`](https://hexdocs.pm/triplex/Triplex.html#all/1) with [`Triplex.migrate`](https://hexdocs.pm/triplex/Triplex.html#migrate/2) and migrate all of the tenants on the database on your remote console.

**Comment by stueccles** [comment]
@kelvinst it would be nice if
```
Enum.each(Triplex.all, &Triplex.migrate(&1) end)
```
worked but alas it doesn't as Triplex.all returns a list of prefixes which Triplex.migrate then applies `to_prefix` on again...

it is not difficult to work around and running 
```
Enum.each(Triplex.all, &Ecto.Migrator.run(
          My.Repo,
          Triplex.migrations_path(My.Repo),
          :up,
          all: true,
          prefix: &1
        ))
```
works in a production terminal to do the tenant migrations.

**Comment by kelvinst** [comment]
Yeah, we actually need to think of better ways of doing those. Maybe a `Triplex.migrate_all` function would make it easier, but we still have to come up with a better way to decide when to apply `to_prefix` and when not to apply it. I just can’t figure any way of doing that without breaking the API so hard that we would need to bump the version to 2.0.

Best,
Kelvin Stinghen
kelvin.stinghen@me.com

**Comment by kelvinst** [comment]
We’re open to suggestions though.

Best,
Kelvin Stinghen
kelvin.stinghen@me.com

**Comment by kelvinst** [comment]
Anyway, we are currently working on a braking change that would force some people to add another lib to their projects if they want the plugs, so why not? 🤷‍♂
